llpc/context/llpcCompiler.cpp

```diff
@@ -1536,8 +1541,11 @@ Result Compiler::BuildGraphicsPipeline(const GraphicsPipelineBuildInfo *pipeline
   const bool buildingRelocatableElf =
       relocatableElfRequested && canUseRelocatableGraphicsShaderElf(shaderInfo, pipelineInfo);
 
-  for (unsigned i = 0; i < ShaderStageGfxCount && result == Result::Success; ++i)
-    result = validatePipelineShaderInfo(shaderInfo[i]);
+  for (ShaderStage stage : gfxShaderStages()) {
+    if (result != Result::Success)
+      break;
```

Comment:
Again, it might read better to place the break after assigning a value to result.